### DISCUSSION TITLE
Welcome to node-red-contrib-tuya-smart-device Discussions!

### DISCUSSION THREAD

**Discussion by vinodsr** [vinodsr/node-red-contrib-tuya-smart-device#32] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

  📢 ** You can use discussions, without starting an issue.

  ➡️ You can **convert issues to discussions** either individually or bulk by labels. Looking at you, issues labeled “question” or “discussion”.

**Comment by Sc00bs110** [comment]
Thank you for the great addon, a sample flow to show how it works in practice would be very useful for the beginners.

Perhaps switching on/off a tuya switch and querying it for power usage as an example

**Reply by thewaver007** [reply]
Hi, I am struggling with vinodsr's flow. I have a Gosund SP116 smart switch (which is a Tuya switch really) and I just cannot get to turn it on or off. I can get it to work from the Tuya website API (eu.iot.tuya.com) but not from a node red flow. Did you ever get a flow to turn on/off a tuya smart switch. regards, a frustrated beginner....

**Comment by Sungtroll** [comment]
Hi.
I think the ducumentation in improper for node-red users. The documentation links to https://codetheweb.github.io/tuyapi/
I want to see real examples to put into node-red inject node and from function node.

**Reply by vinodsr** [reply]
Documentation is available in the README.md as well as in the node-red documentation for the node. 

You can find the example json @https://github.com/vinodsr/node-red-contrib-tuya-smart-device/blob/master/examples

**Comment by evilsnailofdestruktion** [comment]
Hi
So I do not know what I'm doing. I got node red installed, and the node, and my tuya lights connected. But I dont understand how to format my command so it gets read by the light. Can I use a function node? or an input node? and then what do i do after that? I'm trying to set this up to use the flic buttons to set lights on/white/brightness 0, on/colored/brightness 100 and off. Any help is greatly appreciated!

**Reply by evilsnailofdestruktion** [reply]
i get this error ""TypeError: Cannot create property 'shouldWaitForResponse' on string"

**Comment by vinodsr** [comment]
Could you please share the node red logs

On Thu, 8 Sep 2022 at 6:39 AM, evilsnailofdestruktion <
***@***.***> wrote:

> i get this error ""TypeError: Cannot create property
> 'shouldWaitForResponse' on string"
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vinodsr/node-red-contrib-tuya-smart-device/discussions/32#discussioncomment-3590043>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADQ6OCGT6L2TVET36MMN63V5E4EPANCNFSM4VQ5FARA>
> .
> You are receiving this because you authored the thread.Message ID:
> <vinodsr/node-red-contrib-tuya-smart-device/repo-discussions/32/comments/3590043
> @github.com>
>
-- 
Vinod Surendran R
vinodsr.dev

**Reply by evilsnailofdestruktion** [reply]
i actually got it. i needed to use a function node set to:

msg.payload = { "multiple": true, "data": { '1': true, '2': 'white' , '3': 25 } }
return msg;

so now my lights are working!! Thank you for the node that made this possible!!
